File: unity/Assets/Scripts/Multiplayer/Rooms/RoomCreateResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Response returned when a game room is successfully created.
    /// Contains the room ID and configuration details.
    /// </summary>
    [System.Serializable]
    public class RoomCreateResponse : ApiResponse<ERoomCreateError>
    {
        /// <summary>
        /// The unique ID of the created game room.
        /// </summary>
        public string room_id;

        /// <summary>
        /// The name of the game room.
        /// </summary>
        public string room_name;

        /// <summary>
        /// Whether the room supports realtime communication.
        /// </summary>
        public bool realtime;

        /// <summary>
        /// Whether the player who created the room is the host.
        /// </summary>
        public bool is_host;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomCreateResponse.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a game room is successfully created.
 * Contains the room ID and configuration details.
 */
public class RoomCreateResponse extends ApiResponse {
    @JsonProperty("room_id")
    private String roomId;

    @JsonProperty("room_name")
    private String roomName;

    @JsonProperty("realtime")
    private boolean realtime;

    @JsonProperty("is_host")
    private boolean isHost;

    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public boolean isRealtime() {
        return realtime;
    }

    public void setRealtime(boolean realtime) {
        this.realtime = realtime;
    }

    public boolean isHost() {
        return isHost;
    }

    public void setHost(boolean host) {
        isHost = host;
    }
}
